src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8140,130 +8157,197 @@ void checkWellKnown(WellKnownMember member)
-                            property.OriginalDefinition is { ParameterCount: 1 } original &&
-                            original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
+                        foreach (var candidate in lookupResult.Symbols)
                         {
-                            // note: implicit copy check on the indexer accessor happens in CheckPropertyValueKind
-                            patternSymbol = property;
-                            return true;
+                            if (!candidate.IsStatic &&
+                                candidate is PropertySymbol property &&
+                                IsAccessible(property, syntax, diagnostics) &&
+                                property.OriginalDefinition is { ParameterCount: 1 } original &&
+                                original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
+                            {
+                                var intPlaceholder = new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, Compilation.GetSpecialType(SpecialType.System_Int32)) { WasCompilerGenerated = true };
+                                argumentPlaceholders = ImmutableArray.Create(intPlaceholder);
+
+                                var analyzedArguments = AnalyzedArguments.GetInstance();
+                                analyzedArguments.Arguments.Add(intPlaceholder);
+                                var properties = ArrayBuilder<PropertySymbol>.GetInstance();
+                                properties.AddRange(property);
+                                indexerOrSliceAccess = BindIndexerOrIndexedPropertyAccess(syntax, receiver, properties, analyzedArguments, diagnostics).MakeCompilerGenerated();
+                                properties.Free();
+                                analyzedArguments.Free();
+                                lookupResult.Free();
+                                return true;
+                            }
                         }
                     }
                 }
-            }
-            else if (receiverType.SpecialType == SpecialType.System_String)
-            {
-                Debug.Assert(!argIsIndex);
-                // Look for Substring
-                var substring = (MethodSymbol)Compilation.GetSpecialTypeMember(SpecialMember.System_String__Substring);
-                // PROTOTYPE(list-patterns) Consider reporting a missing member for string.Substring
-                if (substring is object)
+                else if (receiver.Type.SpecialType == SpecialType.System_String)
                 {
-                    patternSymbol = substring;
-                    return true;
+                    Debug.Assert(!argIsIndex);
+                    // Look for Substring
+                    var substring = (MethodSymbol)Compilation.GetSpecialTypeMember(SpecialMember.System_String__Substring);
```

Comment:
Should this be the binder version of `GetSpecialTypeMember`?

Reply:
And do we have a test covering the prototype comment?

Reply:
This is covered by `ListPattern_MissingMembers_Substring`.
Good idea, using the binder version of GetSpecialTypeMember yields better diagnostics for that scenario.